Code/Editor/TrackView/TrackViewNodes.cpp

```diff
@@ -2591,11 +2458,108 @@ void CTrackViewNodesCtrl::EndUndoTransaction()
     UpdateDopeSheet();
 }
 
+QIcon CTrackViewNodesCtrl::TrackViewIcon(const CTrackViewTrack* pTrack) {
+    if (!pTrack)
+    {
+        return QIcon(TrackIcon);
+    }
+    const CAnimParamType paramType = pTrack->GetParameterType();
+    const AnimValueType valueType = pTrack->GetValueType();
+    const AnimNodeType nodeType = pTrack->GetAnimNode()->GetType();
+
+    if (nodeType == AnimNodeType::RadialBlur || nodeType == AnimNodeType::ColorCorrection || nodeType == AnimNodeType::DepthOfField ||
+        nodeType == AnimNodeType::ShadowSetup)
+    {
+        return QIcon(TrackIcon);
+    }
+
+    switch (valueType)
+    {
+    case AnimValueType::CharacterAnim:
+    case AnimValueType::AssetBlend:
+        return QIcon(AnimationIcon);
+    }
+
+    AnimParamType type = paramType.GetType();
+    switch (type)
+    {
+    case AnimParamType::FOV:
+        return QIcon(FovIcon);
+    case AnimParamType::Position:
+        return QIcon(PositionIcon);
+    case AnimParamType::Rotation:
+        return QIcon(RotationIcon);
+    case AnimParamType::Scale:
+        return QIcon(ScaleIcon);
+    case AnimParamType::Event:
```

Comment:
add [[fallthrough]]

Reply:
do you need to declare fall-through for this case?

Reply:
Only when there are two cases next to each other without break/return in beetwen

Reply:
that should be fine